app/assets/javascripts/admin/addon/controllers/admin-badges/show.js

```diff
@@ -0,0 +1,238 @@
+import Controller, { inject as controller } from "@ember/controller";
+import { observes } from "discourse-common/utils/decorators";
+import I18n from "I18n";
+import bootbox from "bootbox";
+import { bufferedProperty } from "discourse/mixins/buffered-content";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { next } from "@ember/runloop";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import getURL from "discourse-common/lib/get-url";
+import { tracked } from "@glimmer/tracking";
+
+const IMAGE = "image";
+const ICON = "icon";
+
+// TODO: Stop using Mixin here
+export default class AdminBadgesShowController extends Controller.extend(
+  bufferedProperty("model")
+) {
+  @controller adminBadges;
+  @service router;
+
+  @tracked saving = false;
+  @tracked savingStatus = "";
+  @tracked selectedGraphicType = null;
+
+  get badgeTypes() {
+    return this.adminBadges.badgeTypes;
+  }
+
+  get badgeGroupings() {
+    return this.adminBadges.badgeGroupings;
+  }
+
+  get badgeTriggers() {
+    return this.adminBadges.badgeTriggers;
+  }
+
+  get protectedSystemFields() {
+    return this.adminBadges.protectedSystemFields;
+  }
+
+  get readOnly() {
+    return this.buffered.system;
```

Comment:
When I check out this PR locally and visit the admin badges UI, I get this error:
<img width="871" alt="Screenshot 2022-08-02 at 22 30 16" src="https://user-images.githubusercontent.com/6270921/182477291-2adaff90-34ee-44ca-87a8-a54c60e76c2a.png">

The backtrace points to this `get readOnly` function 👀 

I guess our test coverage here is not great 😢

Reply:
Yeah, `this.buffered` is a proxy that has to be used with `get()`

Reply:
Oops! I forgot to actually look at the output and had just run the tests :/